cli_tools/gce_vm_image_import/importer/inflater.go

```diff
@@ -146,3 +294,146 @@ func getDisk(workflow *daisy.Workflow, diskIndex int) *daisy.Disk {
+		inflater.computeClient.DeleteDisk(inflater.args.Project, inflater.args.Zone, cd.Name)
+	}()
+
+	bkt, objPath, err := storage.GetGCSObjectPathElements(inflater.args.SourceFile)
+	if err != nil {
+		return persistentDisk{}, err
+	}
+	sourceFile := inflater.storageClient.GetObject(bkt, objPath).GetObjectHandle()
+	attrs, err := sourceFile.Attrs(ctx)
+	if err != nil {
+		return persistentDisk{}, daisy.Errf("Failed to get source file attributes: %v", err)
+	}
+	sourceFileSizeGb := (attrs.Size - 1) / 1073741824 + 1
+
+	diskUri := fmt.Sprintf("zones/%s/disks/%s", inflater.args.Zone, diskName)
+	pd := persistentDisk{
+		uri:           diskUri,
+		sizeGb:        cd.SizeGb,
+		sourceGb:      sourceFileSizeGb,
+		sourceType:    "vmdk", // only vmdk is supported right now
+		inflationTime: time.Since(startTime),
+	}
+
+	inflater.addTraceLog("Started checksum calculation.")
+	pd.checksum, err = inflater.calculateChecksum(ctx, diskUri)
+	return pd, err
+}
+
+// run a workflow to calculate checksum, and cleanup the disk
+func (inflater apiInflater) calculateChecksum(ctx context.Context, diskUri string) (string, error) {
+	w := daisy.New()
+	w.Name = "shadow-disk-checksum"
+	checksumScript := checksumScriptConst
+	w.Steps = map[string]*daisy.Step{
+		"create-disks": {
+			CreateDisks: &daisy.CreateDisks{
+				{
+					Disk: compute.Disk{
+						Name:        "disk-${NAME}",
+						SourceImage: "projects/compute-image-tools/global/images/family/debian-9-worker",
+						Type:        "pd-ssd",
+					},
+					FallbackToPdStandard: true,
+				},
+			},
+		},
+		"create-instance": {
+			CreateInstances: &daisy.CreateInstances{
+				Instances: []*daisy.Instance{
+					{
+						Instance: compute.Instance{
+							Name: "inst-${NAME}",
+							Disks: []*compute.AttachedDisk{
+								{Source: "disk-${NAME}"},
+								{Source: diskUri, Mode: "READ_ONLY"},
+							},
+							MachineType: "n1-highcpu-4",
+							Metadata: &compute.Metadata{
+								Items: []*compute.MetadataItems{
+									{
+										Key:   "startup-script",
+										Value: &checksumScript,
+									},
+								},
+							},
+							NetworkInterfaces: []*compute.NetworkInterface{
+								{
+									AccessConfigs: []*compute.AccessConfig{},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"wait-for-checksum": {
+			WaitForInstancesSignal: &daisy.WaitForInstancesSignal{
+				{
+					Name: "inst-${NAME}",
+					SerialOutput: &daisy.SerialOutput{
+						Port:         1,
+						SuccessMatch: "Checksum calculated.",
+						StatusMatch: "Checksum:",
+					},
+				},
+			},
+		},
+	}
+	w.Dependencies = map[string][]string{
+		"create-instance": {"create-disks"},
+		"wait-for-checksum": {"create-instance"},
+	}
+
+	daisycommon.SetWorkflowAttributes(w, inflater.args.Project, inflater.args.Zone, inflater.args.ScratchBucketGcsPath,
+		inflater.args.Oauth, "20m", inflater.args.ComputeEndpoint, inflater.args.GcsLogsDisabled,
+		inflater.args.CloudLogsDisabled, inflater.args.StdoutLogsDisabled)
```

Comment:
I think '20m' means that you're allowing a maximum of 20 minutes to run the checksum workflow; is that right?

Reply:
Yes. Added comment to explain.